### PR TITLE
[ARCTIC-495]remove unnecessary properties when sync and commit properties to ams

### LINKED ISSUE #495: [Improvement]: remove unnecessary properties when sync and commit properties to ams
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?page=4&q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

remove unnecessary properties when sync and commit properties to ams

### How should we improve?

remove unnecessary properties when sync and commit properties to ams

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hameizi**
## Why are the changes needed?
resolve #495 
remove unnecessary properties when sync and commit properties to ams

## Brief change log

remove unnecessary properties when sync and commit properties to ams

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/config/ServerTableProperties.java [hunk 1]
```diff
@@ -47,5 +48,14 @@ public class ServerTableProperties {
     HIDDEN_EXPOSED.add(TableProperties.READ_DISTRIBUTION_MODE_HASH);
     HIDDEN_EXPOSED.add(TableProperties.PROTECTED_PROPERTIES.toString());
     HIDDEN_EXPOSED.add(TableProperties.TABLE_EVENT_TIME_FIELD);
+
+    HIDDEN_INTERNAL.add("meta_store_site");
```
Comment (zhoujinsong) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/config/ServerTableProperties.java:52]: Could change these value with constant variables?

**Review by zhoujinsong — CHANGES_REQUESTED** [review]
@hameizi I left some comments. PTAL.

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/DDLTracerService.java [hunk 2]
```diff
@@ -303,6 +305,8 @@ public void syncDDl(ArcticTable arcticTable) {
 
     public void syncProperties(TableMetadata tableMetadata, ArcticTable arcticTable) {
       Table table = arcticTable.isKeyedTable() ? arcticTable.asKeyedTable().baseTable() : arcticTable.asUnkeyedTable();
+      PropertiesUtil.removeHiddenProperties(tableMetadata.getProperties(), ServerTableProperties.HIDDEN_INTERNAL);
```
Comment (shendanfengg) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/DDLTracerService.java:308]: This map is an immutable map, can not execute remove operation

Comment (hameizi) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/DDLTracerService.java:308]: fixed in new commit

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/config/ServerTableProperties.java (hunk 1 above)
Comment (hameizi) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/config/ServerTableProperties.java:52]: Should we add a CatalogProperties class?

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**